### PR TITLE
Schedule

### PR DESCRIPTION
**Pull request by Martyna-Tu**

### CODE REVIEW

**Comment by ziobron** [comment]
✅ CI Passed - Task: schedule
🏅 Granted 12 XP for @Martyna-Tu

**Comment by ziobron** [comment]
⏱ 3 XP extra for @Martyna-Tu for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: schedule-bonus
🏅 Granted 10 XP for @Martyna-Tu

**Comment by ziobron** [comment]
⏱ 3 XP extra for @Martyna-Tu for on-time delivery

File: homework/schedule/schedule.hpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include <chrono>
+#include <utility>
+
+template <typename Fun, typename... Args>
+void schedule(Fun&& fun,  std::chrono::duration<double> time, Args&&...args) {
+
+        auto startTimer = std::chrono::system_clock::now();
+        auto elapsed_time = [startTimer]() {
+                std::chrono::duration<double> diffrence = std::chrono::system_clock::now() - startTimer;
```
Comment (runer53248) [inline homework/schedule/schedule.hpp:9]: Typo: there should be a "difference". 
Consider returning the result directly:
    return std::chrono::system_clock::now() - startTimer;
The duration can be compared directly (its compare result of count() in operator ==).

**Comment by ziobron** [comment]
Thanks @runer53248 for Code Review.
🏅 1 XP granted

File: homework/schedule/schedule.hpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+                return diffrence.count();
+        };
+
+        while (elapsed_time() < time.count()) {
```
Comment (nauka-programowania-MA) [inline homework/schedule/schedule.hpp:13]: use std::this:thread sleep for or sleep_until instead

File: homework/schedule/schedule.hpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        }
+        fun(std::forward<Args>(args)...);
```
Comment (nauka-programowania-MA) [inline homework/schedule/schedule.hpp:15]: Nice :)

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**